activestorage/lib/active_storage/attached/changes/create_one.rb

```diff
@@ -82,6 +82,18 @@ def find_or_build_blob
           )
         when String
           ActiveStorage::Blob.find_signed!(attachable, record: record)
+        when File, Pathname
+          io = attachable.is_a?(File) ? attachable : attachable.open
+          filename = File.basename(attachable)
+          content_type = Marcel::MimeType.for(io, name: filename)
+
+          ActiveStorage::Blob.create_and_upload!(
+            io: io,
+            filename: filename,
+            content_type: content_type,
+            record: record,
+            service_name: attachment_service_name
+          )
```

Comment:
`create_and_upload!` will try to identify the `content_type` by default.
We can pass `identify: false`, or let ActiveStorage identify the file:
https://github.com/rails/rails/blob/b1ac22fe7edffaf47051c207d2c43d8971b98a25/activestorage/app/models/active_storage/blob.rb#L97-L101

```suggestion
          ActiveStorage::Blob.create_and_upload!(
            io: io,
            filename: filename,
            content_type: content_type,
            record: record,
            identify: false,
            service_name: attachment_service_name
          )
```

Reply:
Sorry, that is not correct, the documentation is incorrect.
If content_type is set it won't identity.